docs/source/tutorials/websockets.rst

```diff
@@ -0,0 +1,1074 @@
+.. _websocket-tutorial:
+
+Websocket Tutorials
+===================
+
+Echo Server Example
+-------------------
+
+An echo server is a simple server that echos any message it receives back to
+the client that sent it.
+
+Code
+~~~~
+
+.. code-block:: text
+   :caption: requirements.txt
+
+   boto3>=1.9.91
+
+
+.. code-block:: python
+   :caption: app.py
+   :linenos:
+
+   from boto3.session import Session
+
+   from chalice import Chalice
+   from chalice import WebsocketDisconnectedError
+
+   app = Chalice(app_name="echo-server")
+   app.websocket_api.session = Session()
+   app.experimental_feature_flags.update([
+       'WEBSOCKETS'
+   ])
+
+
+   @app.on_ws_message()
+   def message(event):
```

Comment:
We should make sure we can handle redeploys if you don't have a connect/disconnect handler.